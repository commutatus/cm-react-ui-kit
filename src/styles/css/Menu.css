.dropdown-parent-card {
  height: 160px;
  width: 250px;
  border-radius: 2px;
  background-color: #FFFFFF;
  box-shadow: 1px 2px 6px 0 rgba(0, 0, 0, 0.1); }
