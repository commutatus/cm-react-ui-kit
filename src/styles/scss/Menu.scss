@import 'variables';
@import 'mixins';
@import 'Global';

.left-top-menu {

}

.right-top-menu {

}   

.menu-container {
    width: 100%;
    padding: 5px 2px;
    box-sizing: border-box;
    .menu-list {
        padding: 7px;
        @include dropdownFont();
        transition: all .3s;
        cursor: pointer;
        &:hover {
            background-color: #F4F5F7;
        }
    }
}

.sub-menu-list {
    position: absolute;
    width: 120px;
    padding: 5px 10px;
    background-color: $white;
    border-radius: $round-2;
    box-shadow: $dropdown-shadow; 
}
