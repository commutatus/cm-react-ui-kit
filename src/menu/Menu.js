import React from 'react'
import '../styles/css/Menu.css'

export default class Menu extends React.Component{
  constructor(props){
    super(props)
    this.state = {}
  }

  handleClick = (e) => {
    let {key, onTitleClick, onClick} = this.props
    if(this.props.onTitleClick){
      onTitleClick({key, domEvent: e})
    }
    if(onClick){
      onClick({key, domEvent: e})
    }
  }

  render() {
    return(
      <div 
        onClick={this._handleClick}
        className={`menu-container ${this.props.className}`}
        style={{...this.props.style}}
      >
        {
          React.Children.map(this.props.children, (child => React.cloneElement(child, {handleClick: this.handleClick})))
        }
      </div>
    )
  }
}
