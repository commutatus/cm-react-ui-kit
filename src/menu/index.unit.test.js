import React from 'react';
import Menu from '../menu';
import {mount} from 'enzyme'

test('Menu rendering properly...', () => {
  const tree = mount(
    // <Menu>
    <Menu.Item>
      Hello brother
    </Menu.Item>
  //   <Menu.Item>
  //     Hello brother
  //   </Menu.Item>
  //   <Menu.Item>
  //     Hello brother
  //   </Menu.Item>
  //   <Menu.SubItem title={"hover me..."}>
  //     <Menu.Item>
  //       Hello brother
  //     </Menu.Item>
  //   </Menu.SubItem>
  // </Menu>
  );
  console.log(tree)
  // let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})