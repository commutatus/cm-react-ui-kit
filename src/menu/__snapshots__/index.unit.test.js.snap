// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu rendering properly... 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Item>
    Hello brother
  </Item>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Hello brother",
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="menu-list"
      >
        Hello brother
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Hello brother",
        "className": "menu-list",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "Hello brother",
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Hello brother",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="menu-list"
        >
          Hello brother
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Hello brother",
          "className": "menu-list",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          "Hello brother",
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
